### DISCUSSION TITLE
Confused about discover and series usage

### DISCUSSION THREAD

**Discussion by minh2134** [Flexget/Flexget#4019] [Q&A] [answered]
I am trying to use flexget with discover and series plugin but flexget always produced no results. Im not sure if I fully understand how this works.
Flexget config.yml:
```
web_server:
  bind: 0.0.0.0
  port: 5050
  web_ui: yes

tasks:
  anime:
    discover:
      what:
        - next_series_episodes:
            from_start: yes
      from:
        - torznab:
            website: 'http://jackett/api/v2.0/indexers/all/results/torznab/'
            apikey: <APIKEY>
            searcher: tv
            categories:
              - 127720
      interval: 1 hour
    
    series:
      - series 1
      - series 2
      
    rtorrent:
      uri: 'http://rtorrent-rutorrent:8000'
      custom1: flexget
      custom2: anime
      mkdir: yes
      path: /downloads/media/anime
```
the log:
```
root@4ef92a22b037:/# flexget execute --discover-now
There is a FlexGet process already running for this config, sending execution there.

2024-07-15 06:07:31 VERBOSE  discover      anime           Discovering 2 titles ...
2024-07-15 06:07:31 INFO     discover      anime           Ignoring interval because of --discover-now
2024-07-15 06:07:31 VERBOSE  details       anime           Task didn't produce any entries.
2024-07-15 06:07:31 VERBOSE  details       anime           Summary - Accepted: 0 (Rejected: 0 Undecided: 0 Failed: 0)
```

If I tried to use another input plugin on top it works but it seems like the discover plugin is ignored entirely.

**Comment by BrutuZ** [comment]
The 4 log lines indicate discovery is working (`Discovering 2 titles`), it just isn't finding anything with the provided queries

**Reply by minh2134** [reply]
In that case is there anything else I can do to get what discover plugin actually search for?
Because I checked the Jackett link and API, all seems fine, and when I do a manual search on Jackett using the series full name it also works.

**Comment by minh2134** [comment]
ok so a bit more digging and I found that during the initialization of the series, flexget asks tvmaze to identify the series (while I did not specify it), then basically just said "nope, can't find any episodes even though we did find the title" if my series has `identified_by` variable set to `sequence` instead of `ep`. When I set it to `ep` it would work, but the particular group I want to find use sequence numbering so it does not show on the search result. Any help?

**Reply by BrutuZ** [reply]
> my series has identified_by variable set to `sequence` instead of `ep`

Nothing weird there, it even says so in the [documentation](https://flexget.com/Plugins/next_series_episodes)
![image](https://github.com/user-attachments/assets/26031e12-a980-47b5-b0ab-666524e7eb7e)

**Reply by BrutuZ** [reply, marked as answer]
> flexget asks tvmaze to identify the series 

That's the [default estimation](https://flexget.com/Plugins/discover#options), if you disable it with `release_estimations: ignore`, then it will even search for `sequence` with `from_start`.

**Reply by minh2134** [reply]
Yeah when i changed series to `sequence` instead of `ep`, i removed the `from_start` option as per wiki said.

Thanks, the default estimation was exactly what i looking for! I'll mark your comment as answer but one additional question: if I want change the lookup source, say, to tvdb, or imdb, or anidb (which afaik doesn't exist yet but still), what config would i put? I tried to use`imdb_lookup: yes` in the task, but it seems like that doesn't affect it.

**Reply by BrutuZ** [reply]
Lookups only affect accepted entries. You'll have to check their fields on output. Something like the `--dump accepted eval` execution flag will show you what you'll have available to use on other actions.

IMDB is bad juju IMHO, they don't offer a free API so the the plugin breaks every now and then when they change the pages and the scraper needs to be tweaked.

**Reply by minh2134** [reply]
Thanks. And you're right, I don't really trust imdb, or even tvdb if it's for anime, it's just the first example i can think of on top of my head atm.

**Reply by BrutuZ** [reply]
You'll have to settle for TVDB/TMDB and their antics though, most anime DBs don't really provide episode metadata so no point in having a lookup.
AFAIK only AniDB provides some metadata for some episodes but their API is... odd. IIRC it uses UDP calls, there's a strict rate-limit and I even believe they'd rather you download their database dump once a week.

There is a AniDB input plugin though, as well as MAL and AniList, you can forward some of their data via the [`configure_` fields](https://flexget.com/Plugins/configure_series#cross-task-settings) of `configure_series`
